### PR TITLE
Modernize Dockerfile; Create Github package

### LINKED ISSUE #102: Official Docker Hub or Github Container Registry images?
Given the heavy build here, I think it would be really beneficial to have an official image on Docker Hub or GHCR? I know Docker is free for public projects and is easy to set up so it rebuilds with every commit.

### PR DESCRIPTION
**Pull request by djdembeck**
- Update Alpine to `3.12`
- Update FFMPEG to `4.3`
- Better readability of Dockerfile
- Package will be created as `m4b-tool` in Github Package Repository

Test package is available on my fork (you will need to run a `docker login` first):
`docker run --rm docker.pkg.github.com/djdembeck/m4b-tool/m4b-tool:latest`

Closes #102

Not sure what https://github.com/sandreas/m4b-tool/blob/8c0aa867d48b5ec238679497d78152b3b2a455dd/Dockerfile#L84 purpose is, so I stuck to just installing prerelease.

### CODE REVIEW

**Comment by sandreas** [comment]
Thank you for providing this pull request. It'll take some time for the review, because I'm not too familiar with github actions but i plan to take a look.

**Comment by djdembeck** [comment]
Of course! I used the default docker template, only changing image name. Essentially, the action will run a test docker build, and if that succeeds, it will build the image, and push to the Github Container repo, as `latest`. If there's anything else I can help with, let me know!

File: Dockerfile [hunk 1]
```diff
@@ -66,38 +70,28 @@ RUN echo "---- INSTALL RUNTIME PACKAGES ----" && \
...
-       fi \
-    && mv /tmp/m4b-tool.phar /usr/local/bin/m4b-tool \
-    && M4B_TOOL_PRE_RELEASE_LINK=$(wget -q -O - https://github.com/sandreas/m4b-tool/releases/tag/latest | grep M4B_TOOL_DOWNLOAD_LINK | cut -d '=' -f 2 | cut -d ' ' -f 1) \
-    && wget "${M4B_TOOL_PRE_RELEASE_LINK}" -O /tmp/m4b-tool.tar.gz \
-    && tar xzf /tmp/m4b-tool.tar.gz -C /tmp/ && rm /tmp/m4b-tool.tar.gz \
-    && mv /tmp/m4b-tool.phar /usr/local/bin/m4b-tool-pre \
-    && chmod +x /usr/local/bin/m4b-tool /usr/local/bin/m4b-tool-pre
 
-WORKDIR ${WORKDIR}
+RUN echo "---- INSTALL M4B-TOOL ----" && \
```
Comment (sandreas) [inline Dockerfile:88]: This part is missing some things:
- A custom local build, that is copied into the image, if present (this is important for my personal tests)
- The process will only install the latest pre-release, while it before installed both the latest pre-release as `m4b-tool-pre` AND the stable release as `m4b-tool`

**Review by sandreas — COMMENTED** [review]
All in all impressive work, but there is unfortunately missing a minor detail (see comments). 
And a question:
Does the `.github/workflows/docker-publish.yml` work without any credentials?

File: Dockerfile (hunk 1 above)
Comment (djdembeck) [reply Dockerfile:88]: I will add these back in then. Sorry for the delay!

**Comment by djdembeck** [comment]
Yes, it will publish without credentials in the pipeline. I do need to update this all for modern changes before a new review.

**Pull request closed without merging**